sqldelight-gradle-plugin/src/test/fixtures/well-formed-selects/expected/com/sample/HockeyPlayerModel.java

```diff
@@ -191,60 +191,248 @@
     TeamModel team();
   }
 
-  interface Select_allCreator<T extends Select_allModel> {
+  interface Select_allCreator<T extends HockeyPlayerModel> {
     T create(HockeyPlayerModel hockey_player, TeamModel team);
   }
 
+  final class Select_allMapper<T extends HockeyPlayerModel, R0 extends HockeyPlayerModel, R1 extends TeamModel> implements RowMapper<T> {
+    private final Select_allCreator<T> creator;
+
+    private final Factory<R0> hockeyPlayerModelFactory;
+
+    private final TeamModel.Factory<R1> teamModelFactory;
+
+    private Select_allMapper(Select_allCreator<T> creator, Factory<R0> hockeyPlayerModelFactory, TeamModel.Factory<R1> teamModelFactory) {
+      this.creator = creator;
+      this.hockeyPlayerModelFactory = hockeyPlayerModelFactory;
+      this.teamModelFactory = teamModelFactory;
+    }
+
+    @Override
+    @NonNull
+    public T map(@NonNull Cursor cursor) {
+      return creator.create(
+          hockeyPlayerModelFactory.creator.create(
+            cursor.getLong(0),
+            cursor.getString(1),
+            cursor.getString(2),
+            cursor.getInt(3),
+            cursor.isNull(4) ? null : cursor.getLong(4),
+            cursor.getInt(5),
+            cursor.getFloat(6),
+            hockeyPlayerModelFactory.birth_dateAdapter.map(cursor, 7),
+            hockeyPlayerModelFactory.shootsAdapter.map(cursor, 8),
+            hockeyPlayerModelFactory.positionAdapter.map(cursor, 9)
+          ),
+          teamModelFactory.creator.create(
+            cursor.getLong(10),
+            cursor.getString(11),
+            teamModelFactory.foundedAdapter.map(cursor, 12),
+            cursor.getString(13),
+            cursor.isNull(14) ? null : cursor.getLong(14),
+            cursor.getInt(15) == 1
+          )
+      );
+    }
+  }
+
   interface For_teamModel {
     HockeyPlayerModel hockey_player();
 
     TeamModel team();
   }
 
-  interface For_teamCreator<T extends For_teamModel> {
+  interface For_teamCreator<T extends HockeyPlayerModel> {
     T create(HockeyPlayerModel hockey_player, TeamModel team);
```

Comment:
I guess this one doesn't matter because the `For_teamModel` doesn't use subtypes either.